### PR TITLE
Remove mapping step from inference pipeline flow

### PR DESCRIPTION
**Pull request by kderusso**
Removes the mapping stage in pipeline inference creation. 

Before: 
before

After: 
after

### CODE REVIEW

File: x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx [hunk 1]
```diff
@@ -33,7 +33,6 @@ import { NoModelsPanel } from './no_models';
 import { ReviewPipeline } from './review_pipeline';
 import { TestPipeline } from './test_pipeline';
 import { AddInferencePipelineSteps } from './types';
-import { UpdateMappings } from './update_mappings';
```
Comment (demjened) [inline x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx:36]: @kderusso Please also delete the component (`update_mappings.tsx`), the unit tests, and the referenced i18n keys from the translation files.

Ideally we should also remove `docLinks.mlDocumentEnrichmentUpdateMappings` which is not used anywhere else in order to not leave it lingering around; it needs to be removed from 4 places if I remember correctly (see [commit that added these](https://github.com/elastic/kibana/commit/8706702aea16ddade385771c71ca3a2a63c45caa)). But if it's too complicated, let's leave it.

File: x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx [hunk 2]
```diff
@@ -225,24 +214,6 @@ describe('AddInferencePipelineFlyout', () => {
...
       testStepStatus(FIELDS_STEP_INDEX, 'Fields', 'complete');
     });
-    it('mappings step is current when on step', () => {
-      setMockValues({
-        ...DEFAULT_VALUES,
-        addInferencePipelineModal: {
-          step: AddInferencePipelineSteps.Mappings,
-        },
-      });
-      testStepStatus(MAPPINGS_STEP_INDEX, 'Mappings', 'current');
```
Comment (demjened) [inline x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx:235]: Please remove `MAPPINGS_STEP_INDEX` and update the ordinals of the other `_INDEX` constants.

**Review by demjened — APPROVED** [review]
LGTM with comments

**Comment by Mikep86** [comment]
Let's make sure to not actually merge this PR (at least not yet) so that we don't mess with the upstream PR

File: x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx (hunk 1 above)
Comment (kderusso) [reply x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx:36]: Done, thanks for the catch!

File: x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx (hunk 2 above)
Comment (kderusso) [reply x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_inference_pipeline_flyout.test.tsx:235]: Done, thanks for the catch!

**Comment by kderusso** [comment]
Closed in favor of: https://github.com/elastic/kibana/pull/171690

**Pull request closed without merging**